react-native-community/cli
# fix: correctly list ios devices and simulators

Summary:
---------

Switched from `xcrun xctrace list devices` to `xcrun xcdevice list` when gathering info about available iOS devices and simulators. 
Previous one wasn't providing enough information to distinguish between AppleTV / macOS / iPhone devices leading to unexpected behaviours. 

Fixes #1816 
Fixes #1807 
Closes #1786 


Test Plan:
----------
Connect as many Apple devices as possible and:
-  try to run `run-ios` - it should try to launch RN App on all valid devices and simulators
- try to run `run-ios --list-devices` - it should list all valid devices (even not available ones) and simulators

<img width="847" alt="image" src="https://user-images.githubusercontent.com/13610886/217224144-877872ef-77e6-4e9d-9685-455244eb047e.png">
